tools/generate_release_notes.py

```diff
@@ -1,246 +1,331 @@
 
-GH_USER = "scikit-image"
+here = Path(__file__).parent
+
+GH_URL = "https://github.com"
+GH_ORG = "scikit-image"
 GH_REPO = "scikit-image"
-GH_TOKEN = os.environ.get("GH_TOKEN")
-if GH_TOKEN is None:
-    raise RuntimeError(
-        "It is necessary that the environment variable `GH_TOKEN` "
-        "be set to avoid running into problems with rate limiting. "
-        "One can be acquired at https://github.com/settings/tokens.\n\n"
-        "You do not need to select any permission boxes while generating "
-        "the token."
-    )
 
-g = Github(GH_TOKEN)
-repository = g.get_repo(f"{GH_USER}/{GH_REPO}")
 
+def lazy_tqdm(*args, **kwargs):
+    """Defer initialization of progress bar until first item is requested."""
+    kwargs["file"] = kwargs.get("file", sys.stderr)
+    yield from tqdm(*args, **kwargs)
+
+
+def commits_between(repo: Repository, start_rev: str, stop_rev: str) -> set[Commit]:
+    """Fetch commits between two revisions excluding the commit of `start_rev`."""
+    # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits
+    comparison = repo.compare(base=start_rev, head=stop_rev)
+    return set(comparison.commits)
+
+
+def pull_requests_from_commits(commits: Iterable[Commit]) -> set[Commit]:
+    """Fetch pull requests that are associated with the given `commits`."""
+    all_pull_requests = set()
+    for commit in commits:
+        commit_pull_requests = list(commit.get_pulls())
+        if len(commit_pull_requests) != 1:
+            logger.info(
+                "commit %s with no or multiple PR(s): %r",
+                commit.html_url,
+                [p.html_url for p in commit_pull_requests],
+            )
+        if any(not p.merged for p in commit_pull_requests):
+            logger.error(
+                "commit %s with unmerged PRs: %r",
+            )
+        for pull in commit_pull_requests:
+            if pull in all_pull_requests:
+                # May happen if
+                logger.info(
+                    "pull request associated with multiple commits: %r",
+                    pull.html_url,
+                )
+        all_pull_requests.update(commit_pull_requests)
+    return all_pull_requests
+
+
+def contributors(
+    commits: Iterable[Commit], pull_requests
+) -> tuple[set[NamedUser], set[NamedUser]]:
+    """Fetch code authors and reviewers.
+
+    `authors` are first authors of commits; co-authors are not included (yet).
+    `reviewers` are users, who added reviews to a merged pull request or create the
+    merge commit for one.
+    """
+    authors = set()
+    reviewers = set()
 
-parser = argparse.ArgumentParser(usage=__doc__)
-parser.add_argument("from_commit", help="The starting tag.")
-parser.add_argument("to_commit", help="The head branch.")
-parser.add_argument(
-    "--version", help="Version you're about to release.", default="0.15.0"
-)
+    for commit in commits:
+        # TODO include co-authors?
```

Comment:
Shall we decide if we want this or not? I think we should honor `co-authored-by`.

Reply:
I think so too. But it's not trivial to merge a list of commit authors and co-authors. 

For commit authors we get login handle (nice unique key) and name from GitHub's API. For co-authors, I couldn't find a way to get them via the API, so I would have to fall back to parsing the commit message which only yields a name and email pair. How would we merge this list with the one based on login-handle as the unique key?

Also I don't think we should display email addresses in our release notes.

Reply:
We could just include it as another list, match by what email addresses we know, and just display the name and handle + link if available. It wouldn't be perfect but better than nothing.

Reply:
See https://github.com/scikit-image/scikit-image/pull/6961/commits/0485601ebe08d4efdda6a60c3ae78096f7d181b3 for a first attempt. It falls back to linking to a random co-authored commit if the user cannot be determined. E.g. changelog since v0.20.0 here:

<details>
<summary>Click to expand</summary>

```markdown
# scikit-image 0.0.0 release notes

We're happy to announce the release of scikit-image 0.0.0!
scikit-image is an image processing toolbox for SciPy that includes algorithms
for segmentation, geometric transformations, color space manipulation,
analysis, filtering, morphology, feature detection, and more.

For more information, examples, and documentation, please visit our website:
https://scikit-image.org

## Highlights

## New Features
- Implement Fisher vectors in scikit-image
([#5349](https://github.com/scikit-image/scikit-image/pull/5349)).
- Add support for y-dimensional shear to the AffineTransform
([#6752](https://github.com/scikit-image/scikit-image/pull/6752)).
- Add new image sequence `skimage.data.palisades_of_vogt` showing in-vivo tissue of the palisades of Vogt
([#6858](https://github.com/scikit-image/scikit-image/pull/6858)).

## Enhancements
- Make join_segmentations return array maps from output to input labels
([#6786](https://github.com/scikit-image/scikit-image/pull/6786)).
- Add support for y-dimensional shear to the AffineTransform
([#6752](https://github.com/scikit-image/scikit-image/pull/6752)).
- Bounding box crop
([#5499](https://github.com/scikit-image/scikit-image/pull/5499)).
- Enable use of `rescale_intensity` with dask array
([#6910](https://github.com/scikit-image/scikit-image/pull/6910)).
- Check if `spacing` parameter is tuple in `regionprops`
([#6907](https://github.com/scikit-image/scikit-image/pull/6907)).

## Performance
- Speed up threshold_local function by fixing call to _supported_float_type
([#6847](https://github.com/scikit-image/scikit-image/pull/6847)).
- Add lazy loading to skimage.color submodule
([#6967](https://github.com/scikit-image/scikit-image/pull/6967)).
- Add Lazy loading to skimage.draw submodule
([#6971](https://github.com/scikit-image/scikit-image/pull/6971)).
- Add Lazy loader to skimage.exposure
([#6978](https://github.com/scikit-image/scikit-image/pull/6978)).
- Add lazy loading to skimage.future module
([#6981](https://github.com/scikit-image/scikit-image/pull/6981)).
- Add lazy loading to skimage.feature module
([#6983](https://github.com/scikit-image/scikit-image/pull/6983)).
- Add lazy loading to graph submodule
([#6985](https://github.com/scikit-image/scikit-image/pull/6985)).
- Add lazy loading to measure submodule
([#6999](https://github.com/scikit-image/scikit-image/pull/6999)).
- Add lazy loading to skimage.transform module
([#7009](https://github.com/scikit-image/scikit-image/pull/7009)).

## Bug Fixes
- Fix threshold_li(): prevent log(0) on single-value background
([#6745](https://github.com/scikit-image/scikit-image/pull/6745)).
- shift and normalize data before fitting circle or ellipse
([#6703](https://github.com/scikit-image/scikit-image/pull/6703)).
- Fix copy-paste error in `footprints.diamond` test case
([#6756](https://github.com/scikit-image/scikit-image/pull/6756)).
- Update .devpy/cmds.py to match latest devpy
([#6789](https://github.com/scikit-image/scikit-image/pull/6789)).
- Avoid installation of rtoml via conda in installation guide
([#6792](https://github.com/scikit-image/scikit-image/pull/6792)).
- Raise error in skeletonize for invalid value to method param
([#6805](https://github.com/scikit-image/scikit-image/pull/6805)).
- Sign error fix in measure.regionprops for orientations of 45 degrees
([#6836](https://github.com/scikit-image/scikit-image/pull/6836)).
- Fix returned data type in `segmentation.watershed`
([#6839](https://github.com/scikit-image/scikit-image/pull/6839)).
- Handle NaNs when clipping in `transform.resize`
([#6852](https://github.com/scikit-image/scikit-image/pull/6852)).
- Fix failing regionprop_table for multichannel properties
([#6861](https://github.com/scikit-image/scikit-image/pull/6861)).
- Refactor `_invariant_denoise` to `denoise_invariant`
([#6660](https://github.com/scikit-image/scikit-image/pull/6660)).
- Expose `color.get_xyz_coords` in public API
([#6696](https://github.com/scikit-image/scikit-image/pull/6696)).
- Fix matrix calculation for shear angle in `AffineTransform`
([#6717](https://github.com/scikit-image/scikit-image/pull/6717)).
- Do not allow 64-bit integer inputs; add test to ensure masked and unmasked modes are aligned
([#6875](https://github.com/scikit-image/scikit-image/pull/6875)).
- Showcase pydata-sphinx-theme
([#6714](https://github.com/scikit-image/scikit-image/pull/6714)).
- Fix typo in apply_parallel introduced in #6876
([#6881](https://github.com/scikit-image/scikit-image/pull/6881)).
- Fix and refactor `deprecated` decorator to `deprecate_func`
([#6594](https://github.com/scikit-image/scikit-image/pull/6594)).
- Fix LPI filter for data with even dimensions
([#6883](https://github.com/scikit-image/scikit-image/pull/6883)).
- Use legacy datasets without creating a `data_dir`
([#6886](https://github.com/scikit-image/scikit-image/pull/6886)).
- Corrected energy calculation in chan vese
([#6902](https://github.com/scikit-image/scikit-image/pull/6902)).
- Raise error when source_range is not correct
([#6898](https://github.com/scikit-image/scikit-image/pull/6898)).
- Add missing backticks to DOI role in docstring of `area_opening`
([#6913](https://github.com/scikit-image/scikit-image/pull/6913)).
- Fix inclusion of `random.js` in HTML output
([#6935](https://github.com/scikit-image/scikit-image/pull/6935)).
- Fix URL of random gallery links
([#6937](https://github.com/scikit-image/scikit-image/pull/6937)).
- Use context manager to ensure urlopen buffer is closed
([#6942](https://github.com/scikit-image/scikit-image/pull/6942)).
- apply spacing rescaling when computing centroid_weighted
([#6900](https://github.com/scikit-image/scikit-image/pull/6900)).
- Fix sparse index type casting in `skimage.graph._ncut`
([#6975](https://github.com/scikit-image/scikit-image/pull/6975)).

## API Changes
- Make join_segmentations return array maps from output to input labels
([#6786](https://github.com/scikit-image/scikit-image/pull/6786)).
- Refactor `_invariant_denoise` to `denoise_invariant`
([#6660](https://github.com/scikit-image/scikit-image/pull/6660)).
- Unify API on seed keyword for random seeds / generator
([#6258](https://github.com/scikit-image/scikit-image/pull/6258)).
- Expose `color.get_xyz_coords` in public API
([#6696](https://github.com/scikit-image/scikit-image/pull/6696)).
- Unify pseudo-random seeding interface
([#6922](https://github.com/scikit-image/scikit-image/pull/6922)).
- Change geometric transform inverse to property
([#6926](https://github.com/scikit-image/scikit-image/pull/6926)).

## Maintenance
- Unpin scipy dependency
([#6773](https://github.com/scikit-image/scikit-image/pull/6773)).
- Fix copy-paste error in `footprints.diamond` test case
([#6756](https://github.com/scikit-image/scikit-image/pull/6756)).
- Relicense CLAHE code under BSD-3-Clause
([#6795](https://github.com/scikit-image/scikit-image/pull/6795)).
- Update .devpy/cmds.py to match latest devpy
([#6789](https://github.com/scikit-image/scikit-image/pull/6789)).
- Relax reproduce section in bug issue template
([#6825](https://github.com/scikit-image/scikit-image/pull/6825)).
- Specify kernel for ipywidgets
([#6849](https://github.com/scikit-image/scikit-image/pull/6849)).
- Rename devpy to spin
([#6842](https://github.com/scikit-image/scikit-image/pull/6842)).
- Update references to outdated dev.py with spin
([#6856](https://github.com/scikit-image/scikit-image/pull/6856)).
- Speed up threshold_local function by fixing call to _supported_float_type
([#6847](https://github.com/scikit-image/scikit-image/pull/6847)).
- Bump 0.21 removals to 0.22
([#6868](https://github.com/scikit-image/scikit-image/pull/6868)).
- Update pre-commits
([#6870](https://github.com/scikit-image/scikit-image/pull/6870)).
- Update dependencies
([#6869](https://github.com/scikit-image/scikit-image/pull/6869)).
- Add test for radon transform on circular phantom
([#6873](https://github.com/scikit-image/scikit-image/pull/6873)).
- Do not allow 64-bit integer inputs; add test to ensure masked and unmasked modes are aligned
([#6875](https://github.com/scikit-image/scikit-image/pull/6875)).
- Don't use mutable types as default values for arguments
([#6876](https://github.com/scikit-image/scikit-image/pull/6876)).
- Point `version_switcher.json` URL at dev docs
([#6882](https://github.com/scikit-image/scikit-image/pull/6882)).
- Fix and refactor `deprecated` decorator to `deprecate_func`
([#6594](https://github.com/scikit-image/scikit-image/pull/6594)).
- allow trivial ransac call
([#6755](https://github.com/scikit-image/scikit-image/pull/6755)).
- Add back parallel tests that were removed as part of Meson build
([#6884](https://github.com/scikit-image/scikit-image/pull/6884)).
- Use legacy datasets without creating a `data_dir`
([#6886](https://github.com/scikit-image/scikit-image/pull/6886)).
- Remove old doc cruft
([#6901](https://github.com/scikit-image/scikit-image/pull/6901)).
- Temporarily pin imageio to <2.28
([#6909](https://github.com/scikit-image/scikit-image/pull/6909)).
- Unify pseudo-random seeding interface follow-up
([#6924](https://github.com/scikit-image/scikit-image/pull/6924)).
- Use imageio v3 API
([#6764](https://github.com/scikit-image/scikit-image/pull/6764)).
- Use pytest.warn instead of custom context manager
([#6931](https://github.com/scikit-image/scikit-image/pull/6931)).
- Follow-up to move to pydata-sphinx-theme
([#6933](https://github.com/scikit-image/scikit-image/pull/6933)).
- Mark functions as `noexcept` to support Cython 3
([#6936](https://github.com/scikit-image/scikit-image/pull/6936)).
- Skip unstable test in `ransac`'s docstring
([#6938](https://github.com/scikit-image/scikit-image/pull/6938)).
- Stabilize EllipseModel fitting parameters
([#6943](https://github.com/scikit-image/scikit-image/pull/6943)).
- Point logo in generated HTML docs at scikit-image.org
([#6947](https://github.com/scikit-image/scikit-image/pull/6947)).
- If user provides RNG, spawn it before deepcopying
([#6948](https://github.com/scikit-image/scikit-image/pull/6948)).
- Skip ransac doctest
([#6953](https://github.com/scikit-image/scikit-image/pull/6953)).
- Make `image_fetcher` and `create_image_fetcher` in `data` private
([#6855](https://github.com/scikit-image/scikit-image/pull/6855)).
- Fix NumPy 1.25 deprecation warnings
([#6969](https://github.com/scikit-image/scikit-image/pull/6969)).
- Expose `GeometricTransform.residuals` in HTML doc
([#6968](https://github.com/scikit-image/scikit-image/pull/6968)).
- Revert jupyterlite
([#6972](https://github.com/scikit-image/scikit-image/pull/6972)).
- Don't test numpy nightlies due to transcendental functions issue
([#6973](https://github.com/scikit-image/scikit-image/pull/6973)).
- Ignore tight layout warning from matplotlib pre-release
([#6976](https://github.com/scikit-image/scikit-image/pull/6976)).
- Remove temporary constraint <2.28 for imageio
([#6980](https://github.com/scikit-image/scikit-image/pull/6980)).
- Don't test numpy prerelease on azure
([#6996](https://github.com/scikit-image/scikit-image/pull/6996)).
- Drop Python 3.8 support per SPEC 0
([#6990](https://github.com/scikit-image/scikit-image/pull/6990)).
- Upper pin imageio
([#7002](https://github.com/scikit-image/scikit-image/pull/7002)).
- Update core dependencies per SPEC0
([#7005](https://github.com/scikit-image/scikit-image/pull/7005)).
- meson: allow proper selection of NumPy, Pythran in cross builds
([#7003](https://github.com/scikit-image/scikit-image/pull/7003)).
- Unpin imageio and add warningfilter
([#7006](https://github.com/scikit-image/scikit-image/pull/7006)).
- Update to latest attach-next-milestone action
([#7014](https://github.com/scikit-image/scikit-image/pull/7014)).

## Documentation
- Merge duplicate instructions for setting up build environment
([#6770](https://github.com/scikit-image/scikit-image/pull/6770)).
- DOC: Fix underline length in `docstring_add_deprecated`
([#6778](https://github.com/scikit-image/scikit-image/pull/6778)).
- Add docstring to `skimage.color` module
([#6777](https://github.com/scikit-image/scikit-image/pull/6777)).
- Fix conda instructions for dev env setup
([#6781](https://github.com/scikit-image/scikit-image/pull/6781)).
- Link full license to README
([#6779](https://github.com/scikit-image/scikit-image/pull/6779)).
- Update docstring in skimage.future module
([#6782](https://github.com/scikit-image/scikit-image/pull/6782)).
- Handle pip-only dependencies when using conda
([#6806](https://github.com/scikit-image/scikit-image/pull/6806)).
- Fix returned data type in `segmentation.watershed`
([#6839](https://github.com/scikit-image/scikit-image/pull/6839)).
- Update references to outdated dev.py with spin
([#6856](https://github.com/scikit-image/scikit-image/pull/6856)).
- Remove outdated build instructions from README
([#6788](https://github.com/scikit-image/scikit-image/pull/6788)).
- Add docstring to the `transform` module
([#6797](https://github.com/scikit-image/scikit-image/pull/6797)).
- Added example to AffineTransform class
([#6859](https://github.com/scikit-image/scikit-image/pull/6859)).
- Added examples to the EssentialMatrixTransform class and its estimation function
([#6832](https://github.com/scikit-image/scikit-image/pull/6832)).
- Update _warps_cy.pyx
([#6867](https://github.com/scikit-image/scikit-image/pull/6867)).
- Document boundary behavior of `draw.polygon` and `draw.polygon2mask`
([#6690](https://github.com/scikit-image/scikit-image/pull/6690)).
- Showcase pydata-sphinx-theme
([#6714](https://github.com/scikit-image/scikit-image/pull/6714)).
- Point `version_switcher.json` URL at dev docs
([#6882](https://github.com/scikit-image/scikit-image/pull/6882)).
- Fix docstring underline lengths
([#6895](https://github.com/scikit-image/scikit-image/pull/6895)).
- Add missing backticks to DOI role in docstring of `area_opening`
([#6913](https://github.com/scikit-image/scikit-image/pull/6913)).
- Simplify installation instruction document
([#6927](https://github.com/scikit-image/scikit-image/pull/6927)).
- Add 0.21 release notes
([#6925](https://github.com/scikit-image/scikit-image/pull/6925)).
- Follow-up to move to pydata-sphinx-theme
([#6933](https://github.com/scikit-image/scikit-image/pull/6933)).
- ENH Add JupyterLite button to gallery examples
([#6911](https://github.com/scikit-image/scikit-image/pull/6911)).
- Update release notes
([#6944](https://github.com/scikit-image/scikit-image/pull/6944)).
- MNT Fix typo in JupyterLite comment
([#6945](https://github.com/scikit-image/scikit-image/pull/6945)).
- Point logo in generated HTML docs at scikit-image.org
([#6947](https://github.com/scikit-image/scikit-image/pull/6947)).
- fix bad link in CODE_OF_CONDUCT.md
([#6952](https://github.com/scikit-image/scikit-image/pull/6952)).
- Expose `GeometricTransform.residuals` in HTML doc
([#6968](https://github.com/scikit-image/scikit-image/pull/6968)).
- Add missing PRs to release notes
([#6949](https://github.com/scikit-image/scikit-image/pull/6949)).
- Use a more descriptive title for current inpainting example
([#6989](https://github.com/scikit-image/scikit-image/pull/6989)).
- Fix URLs to the install page
([#6998](https://github.com/scikit-image/scikit-image/pull/6998)).
- Fix equation for Wiener filter in `restoration.wiener`'s docstring
([#6987](https://github.com/scikit-image/scikit-image/pull/6987)).
- Fix missing links in INSTALL.rst and simplify language
([#6984](https://github.com/scikit-image/scikit-image/pull/6984)).
- Edit installation and contributor guidelines
([#6991](https://github.com/scikit-image/scikit-image/pull/6991)).
- Fix URLs that lead to 404 page
([#7008](https://github.com/scikit-image/scikit-image/pull/7008)).

## Infrastructure
- Relax reproduce section in bug issue template
([#6825](https://github.com/scikit-image/scikit-image/pull/6825)).
- Pin to devpy 0.1 tag
([#6816](https://github.com/scikit-image/scikit-image/pull/6816)).
- Prepare CI configuration for merge queue
([#6771](https://github.com/scikit-image/scikit-image/pull/6771)).
- Rename devpy to spin
([#6842](https://github.com/scikit-image/scikit-image/pull/6842)).
- Cleanup cruft in tools
([#6846](https://github.com/scikit-image/scikit-image/pull/6846)).
- Use lazy loader 0.2
([#6844](https://github.com/scikit-image/scikit-image/pull/6844)).
- Update pre-commits
([#6870](https://github.com/scikit-image/scikit-image/pull/6870)).
- Showcase pydata-sphinx-theme
([#6714](https://github.com/scikit-image/scikit-image/pull/6714)).
- Remove `codecov` dependency which disappeared from PyPI
([#6887](https://github.com/scikit-image/scikit-image/pull/6887)).
- Add CircleCI API token; fixes status link to built docs
([#6894](https://github.com/scikit-image/scikit-image/pull/6894)).
- Temporarily pin imageio to <2.28
([#6909](https://github.com/scikit-image/scikit-image/pull/6909)).
- Add PR links to release notes generating script
([#6917](https://github.com/scikit-image/scikit-image/pull/6917)).
- Use official meson-python release
([#6928](https://github.com/scikit-image/scikit-image/pull/6928)).
- Fix inclusion of `random.js` in HTML output
([#6935](https://github.com/scikit-image/scikit-image/pull/6935)).
- Fix URL of random gallery links
([#6937](https://github.com/scikit-image/scikit-image/pull/6937)).
- Respect SPHINXOPTS and add --install-deps flags to `spin docs`
([#6940](https://github.com/scikit-image/scikit-image/pull/6940)).
- Build skimage before generating docs
([#6946](https://github.com/scikit-image/scikit-image/pull/6946)).
- Enable testing against nightly upstream wheels
([#6956](https://github.com/scikit-image/scikit-image/pull/6956)).
- Add nightly wheel builder
([#6957](https://github.com/scikit-image/scikit-image/pull/6957)).
- Run weekly tests on nightly wheels
([#6959](https://github.com/scikit-image/scikit-image/pull/6959)).
- CI: ensure that a "type: " label is present on each PR
([#6960](https://github.com/scikit-image/scikit-image/pull/6960)).
- Add PR milestone labeler
([#6977](https://github.com/scikit-image/scikit-image/pull/6977)).
- Pin milestone labeler to v0.1.0 SHA
([#6982](https://github.com/scikit-image/scikit-image/pull/6982)).
- Ensure existing target directory for `random.js`
([#7015](https://github.com/scikit-image/scikit-image/pull/7015)).
- Assign next milestone only for PRs targeting `main` branch
([#7018](https://github.com/scikit-image/scikit-image/pull/7018)).
- Add missing directories to `spin docs --clean` command
([#7019](https://github.com/scikit-image/scikit-image/pull/7019)).

## Other

## Contributors
35 authors added to this release (alphabetically):
[@DavidTorpey](https://github.com/DavidTorpey),
[@GGoussar](https://github.com/GGoussar),
[@aeisenbarth](https://github.com/aeisenbarth),
[@i-aki-y](https://github.com/i-aki-y),
[@mahamtariq58](https://github.com/mahamtariq58),
[@scott-vsi](https://github.com/scott-vsi),
[Adam J. Stewart (@adamjstewart)](https://github.com/adamjstewart),
[Adeyemi Biola  (@decorouz)](https://github.com/decorouz),
[Ananya Srivastava (@ana42742)](https://github.com/ana42742),
[Andrew J. Hesford (@ahesford)](https://github.com/ahesford),
[Bohumír Zámečník (@bzamecnik)](https://github.com/bzamecnik),
[Carlos Horn (@carloshorn)](https://github.com/carloshorn),
[Daniel Angelov (@23pointsNorth)](https://github.com/23pointsNorth),
[Dipkumar Patel (@immortal3)](https://github.com/immortal3),
[Enrico Tagliavini (@enricotagliavini)](https://github.com/enricotagliavini),
[Eric Prestat (@ericpre)](https://github.com/ericpre),
[Gregory Lee (@grlee77)](https://github.com/grlee77),
[Hayato Ikoma (@hayatoikoma)](https://github.com/hayatoikoma),
[Jake Martin (@jakeMartin1234)](https://github.com/jakeMartin1234),
[Jarrod Millman (@jarrodmillman)](https://github.com/jarrodmillman),
[Juan Nunez-Iglesias (@jni)](https://github.com/jni),
[Kevin MEETOOA (@kevinmeetooa)](https://github.com/kevinmeetooa),
[Lars Grüter (@lagru)](https://github.com/lagru),
[Loïc Estève (@lesteve)](https://github.com/lesteve),
[Marianne Corvellec (@mkcor)](https://github.com/mkcor),
[Mark Harfouche (@hmaarrfk)](https://github.com/hmaarrfk),
[Matthias Bussonnier (@Carreau)](https://github.com/Carreau),
[Matthias Nwt (@matthiasnwt)](https://github.com/matthiasnwt),
[Matus Valo (@matusvalo)](https://github.com/matusvalo),
[Michael Görner (@v4hn)](https://github.com/v4hn),
[Ramyashri Padmanabhakumar (@rum1887)](https://github.com/rum1887),
[Sean Quinn (@seanpquinn)](https://github.com/seanpquinn),
[Stefan van der Walt (@stefanv)](https://github.com/stefanv),
[Tony Reina (@tonyreina)](https://github.com/tonyreina),
[harshitha kolipaka (@harshithakolipaka)](https://github.com/harshithakolipaka),

6 co-authors added to this release (alphabetically):
[Jarrod Millman](https://github.com/scikit-image/scikit-image/commit/ea2757ce3c4516df0a04c7d4788cf1d53d282944),
[Julien Jerphanion (@jjerphan)](https://github.com/jjerphan),
[Lars Grüter](https://github.com/scikit-image/scikit-image/commit/35bf7bab24c57608d1cd90b6b09fb7f444a843d8),
[Marianne Corvellec (@mkcor)](https://github.com/mkcor),
[Stefan van der Walt](https://github.com/scikit-image/scikit-image/commit/7ee3e6e05550548a9fb8097ae775e740f0ab2f56),
[kczimm](https://github.com/scikit-image/scikit-image/commit/8495b071ff81f3f1c5c4d8a44cb0387a81092066),

29 reviewers added to this release (alphabetically):
[@DavidTorpey](https://github.com/DavidTorpey),
[@aeisenbarth](https://github.com/aeisenbarth),
[@i-aki-y](https://github.com/i-aki-y),
[@kzuiderveld](https://github.com/kzuiderveld),
[Adeyemi Biola  (@decorouz)](https://github.com/decorouz),
[Ananya Srivastava (@ana42742)](https://github.com/ana42742),
[Andrew J. Hesford (@ahesford)](https://github.com/ahesford),
[Brigitta Sipőcz (@bsipocz)](https://github.com/bsipocz),
[Carlos Horn (@carloshorn)](https://github.com/carloshorn),
[Cris Luengo (@crisluengo)](https://github.com/crisluengo),
[Dipkumar Patel (@immortal3)](https://github.com/immortal3),
[Enrico Tagliavini (@enricotagliavini)](https://github.com/enricotagliavini),
[Gregory Lee (@grlee77)](https://github.com/grlee77),
[Henry Pinkard (@henrypinkard)](https://github.com/henrypinkard),
[Jarrod Millman (@jarrodmillman)](https://github.com/jarrodmillman),
[Juan Nunez-Iglesias (@jni)](https://github.com/jni),
[Julien Jerphanion (@jjerphan)](https://github.com/jjerphan),
[Kevin MEETOOA (@kevinmeetooa)](https://github.com/kevinmeetooa),
[Lars Grüter (@lagru)](https://github.com/lagru),
[Marianne Corvellec (@mkcor)](https://github.com/mkcor),
[Mark Harfouche (@hmaarrfk)](https://github.com/hmaarrfk),
[Ramyashri Padmanabhakumar (@rum1887)](https://github.com/rum1887),
[Riadh Fezzani (@rfezzani)](https://github.com/rfezzani),
[Sean Quinn (@seanpquinn)](https://github.com/seanpquinn),
[Sebastian Berg (@seberg)](https://github.com/seberg),
[Sebastian Wallkötter (@FirefoxMetzger)](https://github.com/FirefoxMetzger),
[Stefan van der Walt (@stefanv)](https://github.com/stefanv),
[Tony Reina (@tony-res)](https://github.com/tony-res),
[Tony Reina (@tonyreina)](https://github.com/tonyreina),
```

</details>

Reply:
Hrm, yes, not sure how well this works, especially that it duplicates existing contributors. We could de-duplicate based on names in the contributor list as well?

Reply:
In which case we can just tack these names on to the existing list of authors. Linking the username to a commit feels strange, but it wouldn't by odd to say something like "User Lastname in c46fba" with a link to the latter.

Reply:
Okay, I managed to weed out quite a few bugs in matching and handling co-authors.  :pray: